app/exerlog/lib/src/feature/authentication/widgets/google_signin_button.dart

```diff
@@ -0,0 +1,68 @@
+import 'dart:async';
+
+import 'package:exerlog/src/core/base/extensions/context_extension.dart';
+import 'package:flutter/material.dart';
+
+class GoogleSignInButton extends StatefulWidget {
+  GoogleSignInButton({required this.onPressed});
+
+  final FutureOr Function() onPressed;
+
+  @override
+  _GoogleSignInButtonState createState() => _GoogleSignInButtonState();
+}
+
+class _GoogleSignInButtonState extends State<GoogleSignInButton> {
+  bool _isLoading = false;
+
+  @override
+  Widget build(BuildContext context) {
+    return SizedBox(
+      height: 50,
+      width: context.width * .65,
+      child: OutlinedButton(
+        onPressed: () async {
+          setState(() => _isLoading = true);
+          await widget.onPressed();
+          setState(() => _isLoading = false);
+        },
+        child: _isLoading
+            ? Center(
+                child: CircularProgressIndicator(
+                  strokeWidth: 2,
+                  valueColor: AlwaysStoppedAnimation<Color>(Colors.white),
+                ),
+              )
+            : Row(
+                mainAxisSize: MainAxisSize.min,
+                mainAxisAlignment: MainAxisAlignment.center,
+                children: <Widget>[
+                  Image(
+                    image: AssetImage("lib/assets/google-logo.png"),

```

Comment:
Maybe we could gather all the asset paths into a single file so we don't have to write the path everytime and less chance for mistakes, something like:

```
Image(
 image: AssetImage(Assets.googleLogo),
 height: 20,
),
```

Reply:
Done